Veir/IR/Basic.lean

```diff
@@ -473,6 +473,10 @@ def OperationPtr.allocEmpty (ctx : IRContext) (opType : Nat) : Option (IRContext
   let ctx := newOpPtr.set ctx operation
   (ctx, newOpPtr)
 
+set_option linter.unusedVariables false in
+def OperationPtr.dealloc (op : OperationPtr) (ctx : IRContext)
+    (inBounds: op.InBounds ctx := by grind) : IRContext :=
```

Comment:
Maybe add a comment to the PR message on why we need an unused variable here. This is not clear to me.